File: doctor-appointment-frontend/src/features/auth/services/authService.ts
import api from '../../../services/api';
import { LoginRequest, LoginResponse, RegisterRequest } from '../types/authTypes';

export const authService = {
  login: async (data: LoginRequest): Promise<LoginResponse> => {
    const response = await api.post('/api/auth/login', data);
    return response.data;
  },
  
  register: async (data: RegisterRequest): Promise<LoginResponse> => {
    const response = await api.post('/api/auth/register', data);
    return response.data;
  },

  logout: async (): Promise<void> => {
    await api.post('/api/auth/logout');
  }
};

File: doctor-appointment-backend/src/main/java/com/healthcare/appointment/service/AuthService.java
package com.healthcare.appointment.service;

import com.healthcare.appointment.dto.request.LoginRequest;
import com.healthcare.appointment.dto.request.RegisterRequest;
import com.healthcare.appointment.dto.response.LoginResponse;

public interface AuthService {

    LoginResponse register(RegisterRequest request);

    LoginResponse login(LoginRequest request);

    LoginResponse refreshToken(String refreshTokenStr);

    void logout(Long userId);
}
